storybookjs/storybook
# Core: Fix dotenv file loading and add `env` to main.js

Issue: https://github.com/storybookjs/storybook/issues/14103

## What I did
- The envs going into the preview template should be the filtered and combined envs values from loadEnv, not process.env
- Envs are now retreived froma preset
- presets can now use other presets without explicitly passing down `options`

## How to test

- the official storybook should no longer show the warning.
- in files such as these: [preview-head.html](https://github.com/storybookjs/storybook/blob/2ed4b695115e4872fe071012f70df2baf6d42284/examples/official-storybook/preview-head.html) you should now only be able to interpolate a small subset of environment variables. Allowing users to interpolate any value is a risk. Only values found in the .env files and that start with `STORYBOOK_` are allowed.
